jdbc/src/test/java/io/kestra/jdbc/loader/FlowLoader.java

```diff
@@ -0,0 +1,42 @@
+package io.kestra.jdbc.loader;
+
+import io.kestra.core.models.flows.FlowWithSource;
+import io.kestra.core.repositories.FlowRepositoryInterface;
+import io.kestra.core.repositories.LocalFlowRepositoryLoader;
+import io.kestra.core.utils.TestsUtils;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+
+@Singleton
+public class FlowLoader {
+
+    @Inject
+    private FlowRepositoryInterface flowRepository;
+    @Inject
+    protected LocalFlowRepositoryLoader repositoryLoader;
+
+    public void executeWithFlow(List<String> flowPaths, Executor executor)
+        throws Exception {
+
+        for (String path : flowPaths) {
+            TestsUtils.loads(repositoryLoader, Objects.requireNonNull(TestsUtils.class.getClassLoader().getResource(path)));
+        }
+
+        try {
+            executor.execute();
+        } catch (Exception e){
+            throw e;
+        } finally {
+            flowRepository.findAllForAllTenants().forEach(flow -> flowRepository.delete(
+                FlowWithSource.of(flow, "unused")));
+        }
+    }
+
+    public interface Executor {
```

Comment:
Maybe add `@FunctionalInterface` ? 
Or better use Rethrow.RunnableChecked it should cover what you need here